community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -202,6 +202,11 @@ public void startupState( boolean clean )
     static final Header.Reader NO_HEADER = (cursor,length) -> {};
 
     /**
+     * No-op header writer.
+     */
+    static final Consumer<PageCursor> NO_HEADER_WRITER = pc -> {};
```

Comment:
Because `NO_HEADER` live in `GBPTree` as well, together with `NO_MONITOR`.
I think the convenience of having the common no-op implementations of the constructor parameters available directly in the class itself outweigh them being in the more low lever classes to which the more logically belong.

Especially for `NO_HEADER_WRITER` which is a generic `Consumer<PageCursor>` and thus has no obvious "aah, the no-op I'm looking for should obviously be [here]".
For `NO_HEADER`, that obvious place would be `Header` or `Header.Reader` as you pointed out but I rather have them living together.

Wow, wall of text, sorry about that.